ciao-controller/external_ip.go

```diff
@@ -180,10 +180,16 @@ func (c *controller) ListMappedAddresses(tenant *string) []types.MappedIP {
 	return IPs
 }
 
-func (c *controller) MapAddress(poolName *string, instanceID string) (err error) {
+func (c *controller) MapAddress(tenantID string, poolName *string, instanceID string) (err error) {
 	var m types.MappedIP
+	var i *types.Instance
 
-	i, err := c.ds.GetInstance(instanceID)
+	if tenantID == "public" {
+		// we allow the admin to map anyone's instance
```

Comment:
we are inconsistent. We do in many cases, such as deleting an instance. We don't in some cases, such as deleting a workload.